Lib/typing.py

```diff
@@ -236,9 +236,7 @@ def _type_repr(obj):
     # `_collections_abc._type_repr`, which does the same thing
     # and must be consistent with this one.
     if isinstance(obj, type):
-        if obj.__module__ == 'builtins':
-            return obj.__qualname__
-        return f'{obj.__module__}.{obj.__qualname__}'
+        return obj.__fullyqualname__
```

Comment:
You are changing the contract for the method here. This may result in breaking code relying on previous behavior.

Reply:
I think that you misunderstood the API of this attribute. Please check its documentation in `Doc/library/stdtypes.rst`, or just try my PR.